### PR TITLE
fs/smartfs : apply corruption feature

### PR DESCRIPTION
**Pull request by Taejun-Kwon**
It decided that apply this code to only product branch, but to prevent confusion
, Merge from product branch.

### CODE REVIEW

File: os/fs/driver/mtd/smart.c [hunk 1]
```diff
@@ -4887,8 +4887,34 @@ static int smart_ioctl(FAR struct inode *inode, int cmd, unsigned long arg)
 
 	case BIOC_BULKERASE:
 		ret = MTD_IOCTL(dev->mtd, MTDIOC_BULKERASE, 0);
+
+		lldbg("Format Finished\n");
```
Comment (jeongchanKim) [inline os/fs/driver/mtd/smart.c:4891]: If you want to print this always, please use `printf` instead of `lldbg`.

Comment (Taejun-Kwon) [reply os/fs/driver/mtd/smart.c:4891]: I know, application has another log. so it will be displayed. I will change this to fdbg instead of lldbg.

**Review by jeongchanKim — DISMISSED** [review]
LGTM

**Review by jsdosa — APPROVED** [review]
looks nice to me.

File: os/fs/driver/mtd/smart.c [hunk 2]
```diff
@@ -4887,8 +4891,34 @@ static int smart_ioctl(FAR struct inode *inode, int cmd, unsigned long arg)
...
+		}
+#if CONFIG_SMARTFS_ERASEDSTATE == 0xFF
+		header->status = header->status & ~(SMART_STATUS_RELEASED | SMART_STATUS_COMMITTED);
+#else
+		header->status = header->status | SMART_STATUS_RELEASED | SMART_STATUS_COMMITTED;
+#endif
+		offset = sector * dev->mtdBlksPerSector * dev->geo.blocksize + offsetof(struct smart_sect_header_s, status);
+		ret = smart_bytewrite(dev, offset, 1, &header->status);
+		if (ret != 1) {
+			fdbg("Error %d releasing root sector %d\n" - ret, sector);
```
Comment (jeongchanKim) [inline os/fs/driver/mtd/smart.c:4915]: Could you check this log?
`fdbg("Error %d releasing root sector %d\n" - ret, sector);`
I think
`fdbg("Error %d releasing root sector %d\n", -ret, sector);`

**Review by jsdosa — DISMISSED** [review]
Looks good to me.

**Review by sungkyu0 — APPROVED** [review]
Looks Good~

**Review by jeongchanKim — APPROVED** [review]
LGTM

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**